services/wallet/price/price.go

```diff
@@ -0,0 +1,119 @@
+package price
+
+import (
+	"time"
+)
+
+type DataPoint struct {
+	Price     float64
+	UpdatedAt int64
+}
+
+type DataPerTokenAndCurrency = map[string]map[string]DataPoint
+
+type Manager struct {
+	priceProvider Provider
+	priceCache    DataPerTokenAndCurrency
+}
+
+func NewManager(priceProvider Provider) *Manager {
+	return &Manager{
+		priceProvider: priceProvider,
+		priceCache:    make(DataPerTokenAndCurrency),
+	}
+}
+
+func (pm *Manager) FetchPrice(symbol string, currency string) (float64, error) {
+	symbols := [1]string{symbol}
+	currencies := [1]string{currency}
+
+	prices, err := pm.FetchPrices(symbols[:], currencies[:])
+
+	if err != nil {
+		return 0, err
+	}
+
+	return prices[symbol][currency], nil
+}
+
+func (pm *Manager) FetchPrices(symbols []string, currencies []string) (map[string]map[string]float64, error) {
+	result, err := pm.priceProvider.FetchPrices(symbols, currencies)
+	if err != nil {
+		return nil, err
+	}
+
+	pm.updatePriceCache(result)
+
+	return result, nil
+}
+
+func (pm *Manager) getCachedPrices(symbols []string, currencies []string) DataPerTokenAndCurrency {
```

Comment:
Using the same function name as `GetCachedPrices` is confusing. How about calling the other `getAllCachedPrices` or this one `getCachedPricesFor`

Reply:
True, will do!

Reply:
Done